File: server/python/app/infrastructure/mappers/payment_details/personal_name_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.personal_name import PersonalName
from onlinepayments.sdk.domain.personal_name import PersonalName as PersonalNameSdk

class PersonalNameMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[PersonalNameSdk]) -> Optional[PersonalName]:
        if response is None:
            return None

        return PersonalName(
            first_name = response.first_name,
            surname = response.surname,
            title = response.title
        )

File: server/ruby/app/infrastructure/mappers/payment_details/personal_name_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class PersonalNameMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::PersonalName.new(
            first_name: response.first_name,
            surname: response.surname,
            title: response.title
          )
        end
      end
    end
  end
end